packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -239,14 +242,21 @@ export function createResponseState(
     }
   }
   if (bootstrapScripts !== undefined) {
+    const startScriptWithNonce =
+      nonce === undefined
+        ? startScriptSrc
+        : stringToPrecomputedChunk(
```

Comment:
Precomputed chunks are generally constructed in module scope. The idiomatic way of doing this is to break these chunks up in a static-dynamic-static sandwich

Reply:
@gnoff I updated the code to sandwich style 🥪

I had originally based my code on the pre-existing nonce setup for inline scripts: https://github.com/facebook/react/blob/b1b1940ee363facf5b35157d7cae6f6c83495d6c/packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js#L210-L215

If the updated code looks good, should I apply it there as well?

Reply:
This is different. We use this repeatedly when we emit our fizz runtime scripts. The idea with precomputed chunks is you can do the text encoding once (if configured for the host runtime) so we only gain a benefit if the value is reused. pulling the chunk creation to module scope is our way of reusing the chunk which is why I suggested the sandwhich style for your original use case. Here we are getting reuse by referencing this precomputed chunk on each instruction flush but we can't make it in module scope because we embed the nonce within it

Reply:
OK - if I understand correctly, this is because you return `startInlineScript` from `createResponseState` and use it other places. Since we don't have to do that with src/module scripts, it's actually more efficient to construct them the way you suggested, since we can precompile the `nonce=` chunk once inside the module. 👍

That was a tough one to wrap my head around but I'm pretty sure I have it - thank you for your explanations 🙌